tests/azureml/test_image_creation.py

```diff
@@ -390,10 +421,50 @@ def test_execution_script_init_method_attempts_to_load_correct_azure_ml_model(
         assert get_model_path_call_kwargs["version"] == model_version
 
 
-def test_execution_script_run_method_scores_pandas_dataframes_successfully(
-        sklearn_model, model_path):
+def test_execution_script_run_method_scores_pandas_dfs_successfully_when_model_outputs_numpy_arrays(
+        sklearn_model, sklearn_data, model_path):
     mlflow.sklearn.save_model(sk_model=sklearn_model, path=model_path)
 
+    pyfunc_model = mlflow.pyfunc.load_pyfunc(path=model_path)
+    pyfunc_outputs = pyfunc_model.predict(sklearn_data[0])
+    assert isinstance(pyfunc_outputs, np.ndarray)
+
+    model_mock = Mock()
+    model_mock.name = "model_name"
+    model_mock.version = 1
+
+    with TempDir() as tmp:
+        execution_script_path = tmp.path("dest")
+        mlflow.azureml._create_execution_script(
+                output_path=execution_script_path, azure_model=model_mock)
+
+        with open(execution_script_path, "r") as f:
+            execution_script = f.read()
+
+    # Define the `init` and `score` methods contained in the execution script
+    # pylint: disable=exec-used
+    exec(execution_script, globals())
```

Comment:
would it be safer to call eval instead so that we know the execution_script has not dependency on the current state?

Reply:
Sounds good, thanks. Actually now that I am thinking about it, do you know how does azure use it? It probably just imports it. Maybe we should write it out to a file and import it to match the deployment scenario as closely as possible. But if you think exec is good enough I am fine with that too.